core/src/main/java/de/eldoria/bloodnight/mobs/MobSpawner.java

```diff
@@ -44,13 +47,4 @@ void handleSpawnedMob(Entity entity) {
         // TODO Building the mob with changed attributes, equipment, extensions etc is missing.
         coordinator.register(entity, customMob);
     }
-
-    @Override
-    public void run() {
-        while (!nextTick.isEmpty()) {
-            handleSpawnedMob(nextTick.poll());
-        }
-        nextTick.addAll(schedule);
-        schedule.clear();
```

Comment:
Didnt we decided to keep this logic to keep it cleaner and more grouped?

Reply:
I think not that it's more grouped and cleaner to have 2 queues, that are swapping every tick. Also the scheduler for this runnable is started extern and there's no gurantee in this class, that it's always 1 tick. This approach is much shorter und easier to understand. Debugging shouldn't be a problem here, because it doesn't really differ to the former.